server/compose/automation/records_handler.go

```diff
@@ -232,6 +234,17 @@ func (h recordsHandler) each(ctx context.Context, args *recordsEachArgs) (out wf
 		i.filter.NextPage = nil
 		i.buffer, i.filter, err = h.rec.Find(ctx, i.filter)
 
+		// if exclude array is not empty,filter out records from the buffer
+		for _, idTypeValue := range args.Exclude {
+			if idStr, ok := idTypeValue.Get().(string); ok {
+				id := payload.ParseUint64(idStr)
+				excludeRec, index := slice.HasUint64(i.buffer.IDs(), id)
+				if excludeRec {
+					i.buffer = append(i.buffer[:index], i.buffer[index+1:]...)
+				}
+			}
+		}
+
```

Comment:
Clean this up, please

* constant parsing from TypedValue -> string -> uint64 is just not needed
* `slice.HasUint64` is a linear check; this will probably have a bit of impact if the exclude list is long (probably won't be but could be).
* would it be more efficient to fetch records into a separate slice (not `i.buffer`) and then selectively copy the records over (into `i.buffer`)